### PR TITLE
Player compare/progression playlist

### PR DESCRIPTION
**Pull request by Sciguymjm**

### CODE REVIEW

File: backend/blueprints/spa_api/service_layers/player/play_style.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def create_from_id(cls, id_: str, raw=False, rank=None, replay_ids=None, playlis
         )
 
     @staticmethod
-    def create_all_stats_from_id(id_: str, rank=None, replay_ids=None) -> PlayerDataPoint:
+    def create_all_stats_from_id(id_: str, rank=None, replay_ids=None, playlist=None, win=None) -> PlayerDataPoint:
```
Comment (twobackfromtheend) [inline backend/blueprints/spa_api/service_layers/player/play_style.py]: please type.

File: webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx [hunk 2]
```diff
@@ -2,25 +2,35 @@ import { FormControlLabel, Grid, Switch, Typography } from "@material-ui/core"
...
+    playlist: number
+    handlePlaylistChange?: (playlist: number) => void
 }
 
 interface State {
     playerPlayStyles: PlayStyleResponse[]
     playerPlayStylesRaw: PlayStyleRawResponse[]
     rank: number
     heatmapMode: boolean
+    playlist: number
```
Comment (twobackfromtheend) [inline webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx]: playlist in both props and state?

File: webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx [hunk 3]
```diff
@@ -66,6 +79,17 @@ export class PlayerComparePlayStyleCharts extends React.PureComponent<Props, Sta
...
+
+        const dropDown = (
+            <PlaylistSelect
+                selectedPlaylists={[this.props.playlist]}
+                handleChange={this.handlePlaylistsChange}
+                inputLabel="Playlist"
+                helperText="Select playlist to use"
+                dropdownOnly
+                currentPlaylistsOnly
+                multiple={false}/>
```
Comment (twobackfromtheend) [inline webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx]: I think this should be better typed - if `multiple` is false, `selectedPlaylist` (no s!) takes a number and `handleChange` takes a (number => void) whereas if `multiple` is true, `selectedPlaylists` takes `number[]` and handleChange takes (number[] => void)

File: webapp/src/Components/Player/Compare/Progression/PlayerProgressionCharts.tsx [hunk 4]
```diff
@@ -201,4 +220,11 @@ export class PlayerProgressionCharts extends React.PureComponent<Props, State> {
     private readonly handleTimeUnitChange: React.ChangeEventHandler<HTMLSelectElement> = (event) => {
         this.setState({timeUnit: event.target.value as TimeUnit})
     }
+    private readonly handlePlaylistsChange: React.ChangeEventHandler<HTMLSelectElement> = (event) => {
```
Comment (twobackfromtheend) [inline webapp/src/Components/Player/Compare/Progression/PlayerProgressionCharts.tsx]: blank line above.

File: webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx (hunk 3 above)
Comment (Sciguymjm) [reply webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx]: I'm not sure how to do that lol

File: webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx (hunk 2 above)
Comment (Sciguymjm) [reply webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx]: Old stuff you didn't catch last PR.

File: webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx (hunk 3 above)
Comment (twobackfromtheend) [reply webapp/src/Components/Player/Compare/PlayStyle/PlayerComparePlayStyleCharts.tsx]: a simple way I see to do it is 
```ts
interface MultiplePlaylistsProps {
    multiple: true
    playlists: thing[]
    handle: func
}

interface SinglePlay.. {...}

interface Props = (Single | Multiple)
 & whatever.
```

File: webapp/src/Requests/Player/getPlayStyle.ts [hunk 5]
```diff
@@ -4,11 +4,18 @@ import { doGet } from "../../apiHandler/apiHandler"
...
+                             win?: boolean): Promise<PlayStyleResponse> => {
     if (useMockData) {
         return Promise.resolve(MOCK_PLAY_STYLE)
     }
-    const url = qs.stringify({rank, playlist}, {addQueryPrefix: true, indices: false})
+    let params = {}
+    if (win === undefined) {
+        params = {rank, playlist}
+    } else {
+        params = {rank, playlist, win: win ? 1 : 0}
```
[the remark this answers is not in the document]
Comment (Sciguymjm) [reply webapp/src/Requests/Player/getPlayStyle.ts]: Specifying 3 different scenarios:

- We don't want to filter by wins or losses
- We want to filter by wins
- We want to filter by losses

[the remark this answers is not in the document]
Comment (twobackfromtheend) [reply webapp/src/Requests/Player/getPlayStyle.ts]: I think you should do const params = {rank, playlist}, if (win?pleaserenamethis === undefined) {params.result = win ? "win" : "loss}

**Pull request merged**